src/Symfony/Component/Console/Style/SymfonyStyle.php

```diff
@@ -218,21 +217,10 @@ public function caution($message)
      */
     public function table(array $headers, array $rows)
     {
-        array_walk_recursive($headers, function (&$value) {
-            if ($value instanceof TableCell) {
-                $value = new TableCell(sprintf('<info>%s</>', $value), array(
-                    'colspan' => $value->getColspan(),
-                    'rowspan' => $value->getRowspan(),
-                ));
-            } else {
-                $value = sprintf('<info>%s</>', $value);
-            }
-        });
-
         $table = new Table($this);
         $table->setHeaders($headers);
         $table->setRows($rows);
-        $table->setStyle('symfony-style-guide');
+        $table->setStyle((clone Table::getStyleDefinition('symfony-style-guide'))->setCellHeaderFormat('<info>%s</info>'));
```

Comment:
this syntax cannot be used in Symfony 2.7, because it works only for PHP 5.4+, and Symfony 2.3 still supports PHP 5.3